src/main/java/com/coinbase/core/client/CoinbaseNetHttpClient.java

```diff
@@ -0,0 +1,128 @@
+package com.coinbase.core.client;
+
+import com.coinbase.core.common.HttpMethod;
+import com.coinbase.core.credentials.CoinbaseCredentials;
+import com.coinbase.core.errors.CoinbaseClientException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public abstract class CoinbaseNetHttpClient implements CoinbaseClient {
+    protected static final ObjectMapper mapper = new ObjectMapper();
+    private final HttpClient client;
+    private final CoinbaseCredentials credentials;
+    private final String baseUrl;
+
+    public CoinbaseNetHttpClient(CoinbaseCredentials credentials, String baseUrl) {
```

Comment:
Add a new constructor that doesn't take baseUrl or http client


public CoinbaseNetHttpClient(CoinbaseCredentials credentials)

Reply:
Use the default http client if not passed